app/src/main/java/org/lineageos/glimpse/fragments/SearchFragment.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.lineageos.glimpse.fragments
+
+import android.os.Bundle
+import android.view.View
+import android.widget.LinearLayout
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.search.SearchBar
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.launch
+import org.lineageos.glimpse.R
+import org.lineageos.glimpse.ext.getViewProperty
+import org.lineageos.glimpse.recyclerview.LocationsCarouselAdapter
+import org.lineageos.glimpse.ui.ListItem
+import org.lineageos.glimpse.viewmodels.MediaViewModel
+
+/**
+ * A fragment showing a search bar with locations and categories.
+ * Use the [SearchFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class SearchFragment : Fragment(R.layout.fragment_search) {
+    // View models
+    private val mediaViewModel: MediaViewModel by viewModels { MediaViewModel.Factory }
+
+    // Views
+    private val categoriesLinearLayout by getViewProperty<LinearLayout>(R.id.categoriesLinearLayout)
+    private val locationsCarouselRecyclerView by getViewProperty<RecyclerView>(R.id.locationsCarouselRecyclerView)
+    private val searchBar by getViewProperty<SearchBar>(R.id.searchBar)
+
+    // Fragments
+    private val parentNavController by lazy {
+        requireParentFragment().requireParentFragment().findNavController()
+    }
+
+    // Coroutines
+    private val mainScope = CoroutineScope(Job() + Dispatchers.Main)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val locationsCarouselAdapter = LocationsCarouselAdapter()
+
+        locationsCarouselRecyclerView.adapter = locationsCarouselAdapter
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.RESUMED) {
+                mainScope.launch {
+                    locationsCarouselAdapter.clearLocations()
+                }
+                mediaViewModel.locations.take(10).collect {
+                    mainScope.launch {
+                        locationsCarouselAdapter.addLocation(it)
+                    }
+                }
+            }
+        }
+
+        categoriesLinearLayout.let {
```

Comment:
why not with(categoriesLinearLayout)
I prefer using let only in case the value is nullable, otherwise with() is likely nicer